src/GitScan/GitRepo.php

```diff
@@ -201,6 +201,15 @@ public function getRemoteUrls() {
   }
 
   /**
+   * Get the Origin URL.
+   *
+   * @return string|null
+   */
+  public function getOriginUrl() {
+    return array_key_exists('origin', $this->getRemoteUrls()) ? $this->getRemoteUrls()['origin'] : NULL;
```

Comment:
Instead of hardcoding the reference to `origin`, you could do something like `basename($this->getUpstreamBranch())`?